app/infrastructure/mailing_service/models/context_shapes.py

```
@@ -1,11 +1,46 @@
     """Marker base for all email context models."""
 
-class ProjectAnalysisFailure(BaseContextShape):
+class ProjectAnalysisFailureTemplateLayout(BaseModel):
+    """
+        Fields that control width and spacing internally inside the template, no need to pass it manually when instance creating, just statically change it here
+        - `__template_layout_rail_w` -> `RAIL_W` (gutter width)
+            Think of this as a fixed left “lane” reserved for the tree graphics (the dots, dashed line, and elbows).
+                - Bigger RAIL_W → more breathing room for the tree art; the text starts further to the right.
+                - Smaller RAIL_W → the text starts closer to the left edge.
+                - Use when: long function names or you want the connectors to feel roomy.
+                - Typical range: 120–220 px (default in examples: 120).
```

Ruff (0.13.1)

Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)